presto-main/src/main/java/io/prestosql/metadata/Metadata.java

```diff
@@ -571,4 +573,62 @@ default ResolvedFunction getCoercion(Type fromType, Type toType)
      * The method is used by the engine to determine if a materialized view is current with respect to the tables it depends on.
      */
     MaterializedViewFreshness getMaterializedViewFreshness(Session session, TableHandle tableHandle);
+
+    /**
+     * Redirect to another table or view. Returns the original table name if the redirection doesn't happen.
+     */
+    default QualifiedObjectName redirectTable(Session session, QualifiedObjectName tableName, WarningCollector warningCollector)
+    {
+        return tableName;
```

Comment:
There are 2 implementations of this interface (1 non-test), so no need for a default implementation.